client/app/controllers/collection/add.js

```diff
@@ -83,43 +11,125 @@ export default Ember.Controller.extend({
     }),
 
     widgets: [],
+    actions: [],
+
+    // Fire enabled actions.
+    updateState: function(actions) {
+        actions.forEach((action) => {
+            // Check if the action can fire.
+            const may_fire = check_all.call(this, action.conditions);
+            if (!may_fire) return;
+            // Action may fire if execution has reached this point.
+            // Call the action and set its result and any
+            // changes to its state on `controller.parameters`.
+            console.log('stop here');
+            action.output_parameter['value'] = action.action.apply(this, action.arg_arr);
+            action.output_parameter['state'] = ['defined'];
+        });
+    },
+
+
+    // Take the description of an action and set its properties to be the vaious literal
+    // functions and parameters it depends on to operate.
+    hydrate_action: function(action) {
```

Comment:
Please provide a more descriptive name here that indicates what the function is doing